pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -395,7 +395,10 @@ func (ku *KubeUtil) setupKubeletApiEndpoint() error {
 			log.Debugf("Kubelet endpoint is: %s", ku.kubeletApiEndpoint)
 			return nil
 		}
-		return fmt.Errorf("unexpected status code %d on endpoint %s%s", code, ku.kubeletApiEndpoint, kubeletPodPath)
+		if code != http.StatusUnauthorized {
```

Comment:
Would it make sense to also fallback if we get a 403 / `StatusForbidden` ?

Reply:
we should pretty much cover all client errors 400 <= code < 500